File: newton.py
import math as m

def evaluate(f, x0):
  x = x0
  return eval(f)
  
# f, f', starting point, # of iterations
def newton(f, fp, x0, it):
  for i in range (0, it):
    x1 = x0 - evaluate(f, x0) / evaluate(fp, x0)
    x0 = x1
    print (x0, evaluate(f, x0))
  
# newton's method for e^x - 1.5 - arctan(x) starting at x0 = -7 for 8 iterations
print (newton("m.exp(x) - 1.5 - m.atan(x)", "m.exp(x) - 1 / (1 + x * x)", -7, 8))

File: newton.m
function newton(f, fp, x0, it)
  if (it == 0)
      return;
  end
  x1 = x0 - f(x0) / fp(x0);
  x0 = x1;
  fprintf('%d %d\n', x0, f(x0));
  newton(f, fp, x0, it - 1);
end

% Newton's method implementation in Matlab
% PARAMETERS: FUNCTION, DERIVATIVE OF FUNCTION, STARTING POINT, NUMBER OF ITERATIONS
% sample input: newton(@(x) exp(x) - 1.5 - atan(x), @(x) exp(x) - 1 / (1 + x^2), -7, 8)
